### ISSUE TITLE
no-unused-variable triggered in for loop

### ISSUE THREAD

**Issue by helenanderson** [palantir/tslint#3723]
### Bug Report

- __TSLint version__: 5.9.1
- __TypeScript version__: 2.7.1
- __Running TSLint via__: (pick one) CLI  / VSCode

#### TypeScript code being linted

```ts
const metadataList = [];
for (const char of text) {
    const characterMetadata = MyMetadataObject.create();
    metadataList.push(characterMetadata);
}
```

with `tslint.json` configuration:

```json
{
	"extends": [
		"tslint:all",
		"tslint-microsoft-contrib",
		"tslint-react"
	],
	"rules": {
		"no-unused-variable": true,
                "prefer-for-of": true,
      }
}
```

#### Actual behavior
ERROR: no-unused-variable: 'char' is declared but its value is never read

#### Expected behavior
Seems like there should be an exception to no-unused-variable when a variable is declared in the `for` clause of a loop. There are cases where we need a loop so we need to declare a variable, but we don't actually need to use the value of that variable inside the loop.

**Comment by andy-ms** [comment]
You should be able to write `for (const _ of text)`, or use any name beginning with `_`, like you can for a parameter.

**Comment by JoshuaKGoldberg** [comment]
What if `no_unused_variable` is set to _not_ allow trailing underscores as a variable name?

**Comment by andy-ms** [comment]
> What if no_unused_variable is set to not allow trailing underscores as a variable name?

Huh, do you mean you want an option to disable the behavior of allowing `_foo` to be unused?

**Comment by JoshuaKGoldberg** [comment]
I mean, if `no-unused-variable` is set to not allow an `ignore-pattern` so that all unused variables are flagged regardless of name, this issue still applies.

**Comment by JoshuaKGoldberg** [comment]
Coming back to this @helenanderson - now that `no-unused-variable` is deprecated as of TS 2.9 and just receives its diagnostics from TypeScript, and both it and TypeScript allow a preceding `_` in the variable name, closing this issue. Please do comment if you think TSLint should do something here!
